File: 739-daily-temperatures/daily-temperatures.java
class Pair{
    int element;
    int index;
    public Pair(int element, int index){
        this.element=element;
        this.index=index;
    }
}
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n= temperatures.length;
        int[] result=new int[n];
        Stack<Pair> st=new Stack<>();
        for(int i=n-1; i >=0 ;i--){
            while(!st.isEmpty() && st.peek().element<=temperatures[i]){
                st.pop();
            }
            if(st.isEmpty()){
                result[i]=0;
            }else{
                 result[i]=st.peek().index-i;
            }
            st.push(new Pair(temperatures[i],i));
        }
        return result;
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        n= len(temperatures)
        st=[]
        res=[0]*n
        for i in range(n-1,-1,-1):
            while( st and temperatures[i]>= temperatures[st[-1]]):
                st.pop()

            if st:
                res[i]=st[-1]-i
            else:
                res[i]=0
            st.append(i)
        return res
            

